plugin/trino-hive/src/main/java/io/trino/plugin/hive/projection/Projection.java

```diff
@@ -22,4 +22,9 @@ sealed interface Projection
         permits DateProjection, EnumProjection, InjectedProjection, IntegerProjection
 {
     List<String> getProjectedValues(Optional<Domain> partitionValueFilter);
+
+    default String transforming(String value)
```

Comment:
Maybe rename to: `toPartitionLocationTemplateValue` to make the purpose of the transformed result more clear.